pkg/api/scylla/v1alpha1/types_monitoring.go

```diff
@@ -162,6 +162,15 @@ type ScyllaDBMonitoringSpec struct {
 
 	// components hold additional config for the monitoring components in use.
 	Components *Components `json:"components"`
+
+	// type determines the platform type of the monitoring setup. Can be "Platform" or "SaaS".
+	// "Platform" sets up a detailed ScyllaDB monitoring, including the infrastructure.
+	// "Saas" deploys monitoring setup focusing on the ScyllaDB service.
+	// Defaults to "SaaS".
+	// +kubebuilder:validation:Enum="SaaS";"Platform"
+	// +kubebuilder:default:="SaaS"
+	// +optional
+	Type string `json:"type,omitempty"`
```

Comment:
ScyllaDBMonitoringType instead of string

Reply:
now i see it has both the new and old changes which needs cleanup